### DISCUSSION TITLE
iOS - using server presents as WEB on simulator

### DISCUSSION THREAD

**Discussion by OskarLebuda** [ionic-team/capacitor#6394] [Q&A] [answered]
Hi,
Is anybody notice that when we're using `server.url` in `capacitor.config.ts` and run the app using:
```
$ npx cap run ios
```
`window.Capacitor` shows that we're currently using the `web` platform instead of `ios`?
Here is some example:
``` typescript
...
// run HRM for capacitor only on dev
server: (process.env.NODE_ENV === 'production') ? undefined : {
    url: `http://${resolveUserHost()}:${SERVER_PORT}`,
    cleartext: true,
},
...
```
the `SERVER_PORT` and `resolveUserHost()` resolves the `webpack-dev-server` host which is responsible for managing HMR.

Then when I run the simulator and check devtools:
![image](https://user-images.githubusercontent.com/10373127/225046578-3a92201d-d9c5-4cac-9ae4-02adf4c675b8.png)

Is it correct? I think this is some kind of bug. IMO this should be presented as `ios` platform instead of `web`. But let me know if we're doing something wrong.

The interesting thing is that Android emulator with same configuration presents as `android`.
![image](https://user-images.githubusercontent.com/10373127/225045777-4b89be82-1f58-45e6-aed6-1f2ca70422e8.png)

**Comment by jcesarmobile** [comment, marked as answer]
Are you setting `WKAppBoundDomains` in your app?
I think that can cause Capacitor plugins to use the web implementation.

**Reply by OskarLebuda** [reply]
U mean this?
![image](https://user-images.githubusercontent.com/10373127/225414869-96bbe601-5501-4654-a293-8817a77d5f1c.png)

the `resolveUserHost` function returns user device IP. That's the way to enable HMR. 
Should I add my IP every time when I run the code using server?

**Reply by jcesarmobile** [reply]
yeah, that prevents plugins from working, see how to fix it here
https://capacitorjs.com/docs/v3/ios/troubleshooting#plugin-not-implemented
